src/lib/code-sample-editor.ts

```diff
@@ -25,44 +27,24 @@ export class CodeSampleEditor extends LitElement {
   private lastProjectPath?: string;
   private lastSandboxScope: string|null = null;
   private projectContentsReady: Promise<FileRecord[]> = Promise.resolve([EMPTY_INDEX]);
-  private swSetup:Promise<[ServiceWorker | null, ServiceWorkerRegistration | null]> =
-      Promise.resolve([null, null])
-  private swPortEstablished:Promise<null|MessagePort> = Promise.resolve(null);
+  private remoteSw:Promise<null|Comlink.Remote<SwControllerAPI>> = Promise.resolve(null);
   private sessionId: string = generateUniqueSessionId();
 
   private connectToServiceWorker = ():boolean => {
     if (this.lastSandboxScope !== this.sandboxScope) {
       this.lastSandboxScope = this.sandboxScope;
-      // remove previous event listener
-      this.swPortEstablished.then(port => {
-        if (!port) {
-          return;
-        }
-
-        port.removeEventListener('message', this.onSwMessage);
-        clearSession(this.sessionId, port);
-      });
 
-      this.swSetup = setUpServiceWorker(this.sandboxScope);
-
-      this.swPortEstablished = this.swSetup
-          .then((response) => {
-        const sw = response[0];
+      this.remoteSw = this.remoteSw.then(sw => {
```

Comment:
<🏌️‍♂️>
```ts
this.remoteSw = this.remoteSw
  .then(sw => sw && sw.clearContents(this.sessionId))
  .then(async _ => await setUpServiceWorker(this.sandboxScope));
```
</🏌️‍♂️>

Reply:
had to add async or else TS doesn't like returning `null | PromiseLike<void>`